### PR TITLE
Core tests: stream

### PR DESCRIPTION
**Pull request by XLegland**

### CODE REVIEW

File: core/tests/test_stream.h [hunk 1]
```diff
@@ -0,0 +1,239 @@
...
+	stream_free(s);
+	free(buf);
+	remove(file_name);
+}
+
+// check that the output matches the content
+void test_stream_read(void) {
+	const size_t len = 6;
+	char* content = (char*)malloc(len * sizeof(char));
+	content = "Hello";
```
Comment (gdevillele) [inline core/tests/test_stream.h:58]: it doesn't work like that.
we can talk about it.

If I understand correctly, `strcpy` must be used here.

Also, aren't you getting a warning regarding assigning a `const char*` value to a `char*` variable?

File: core/tests/test_stream.h [hunk 2]
```diff
@@ -0,0 +1,239 @@
...
+
+	stream_free(s);
+	free(buf);
+}
+
+// check that the file is read correctly
+void test_stream_new_file_read(void) {
+	const char* file_name = "hi.txt";
+	const size_t len = 6;
+	const char content[6] = "Hello";
```
Comment (gdevillele) [inline core/tests/test_stream.h:36]: I suppose `content` should be of type `const char*`.

**Review by gdevillele — CHANGES_REQUESTED** [review]
Thanks. Partial review done. Let's fix the first issues here.

File: core/tests/test_stream.h (hunk 1 above)
Comment (XLegland) [reply core/tests/test_stream.h:58]: No warning on mvs 2022

File: core/tests/test_stream.h [hunk 3]
```diff
@@ -0,0 +1,238 @@
...
+	char* buf = (char*)malloc(len * sizeof(char));
+	Stream* s = stream_new_buffer_read(content, len);
+	const bool ok = stream_read(s, buf, len, 1);
+
+	TEST_CHECK(ok);
+	TEST_CHECK(strcmp(buf, content) == 0);
+	// cannot read beyond the stream
+	TEST_CHECK(stream_read(s, buf, 1, 1) == false);
+
+	stream_free(s); // frees content
```
Comment (gdevillele) [inline core/tests/test_stream.h:68]: from what I can see, it does not free `content`.

**Review by gdevillele — CHANGES_REQUESTED** [review]
one last change requested.

File: core/tests/test_list.c [hunk 4]
```diff
@@ -49,6 +50,158 @@ TEST_LIST = {
     {"test_aware_block_set_touched_face", test_aware_block_set_touched_face},
     {"test_block_getNeighbourBlockCoordinates", test_block_getNeighbourBlockCoordinates},
 
+    // filo_list_int3
+    { "filo_list_int3_pop", test_filo_list_int3_pop },
```
Comment (gdevillele) [inline core/tests/test_list.c:54]: this is duplicated (others are duplicated as well)

File: core/tests/test_stream.h [hunk 5]
```diff
@@ -0,0 +1,238 @@
...
+// stream_buffer_unload
+// stream_new_file_write
+
+// check that we read the content correctly
+void test_stream_new_buffer_read(void) {
+	const size_t len = 6;
+	const char* content = "Hello";
+	Stream* s = stream_new_buffer_read(content, len);
+	char* buf = (char*)malloc(len * sizeof(char));
+	stream_read_string(s, len, buf);
```
Comment (gdevillele) [inline core/tests/test_stream.h:24]: return value of `stream_read_string` is not tested

File: core/tests/test_stream.h [hunk 6]
```diff
@@ -0,0 +1,238 @@
...
+	const size_t len = 6;
+	const char* content = "Hello";
+	char* buf = (char*)malloc(len * sizeof(char));
+	FILE* f = fopen(file_name, "w"); // open for writing
+	const int err = fputs(content, f);
+	TEST_ASSERT(err != EOF);
+	fclose(f);
+	f = fopen(file_name, "r");	     // open for reading
+	Stream* s = stream_new_file_read(f);
+	stream_read_string(s, 5, buf);
```
Comment (gdevillele) [inline core/tests/test_stream.h:44]: return value of stream_read_string is not tested

File: core/tests/test_stream.h (hunk 3 above)
Comment (gdevillele) [reply core/tests/test_stream.h:68]: comment `// frees content` must be removed

**Review by gdevillele — CHANGES_REQUESTED** [review]
You can squash this into a single commit.
I added a few comments.

**Pull request merged**